### DISCUSSION TITLE
Why use {% if request.GET.tag|length %}

### DISCUSSION THREAD

**Discussion by fabtjar** [wagtail/wagtail#7598] [General]
Hey, I'm looking at the [getting started](https://docs.wagtail.io/en/stable/getting_started/tutorial.html) docs and in one of the examples it uses `{% if request.GET.tag|length %}`. I'm not sure why `length` is used as `{% if request.GET.tag %}` would work the same way as far as I understand. In what way does `length` help here? I must be missing something about how Django templates work.

**Comment by Scotchester** [comment]
Hey @fabtjar,

Without testing it to confirm, I would guess that it's because you can have GET parameters present that have no meaningful value. For example, if the request was to: `/some/path/?emptyparam=&nonemptyparam=foo`, then I would expect `{% if request.GET.emptyparam %}` to evaluate to `True`, despite it not having been given an actual value.

**Reply by fabtjar** [reply]
Hi @Scotchester, Cheers for the fast reply.

I did try that but `?tag=&nonemptyparam=foo` will return `False` with and without `length` being used.

Maybe there could have been an older version of Django when this would have returned `True` though?

**Reply by Scotchester** [reply]
Interesting! It certainly could be a historical artifact like that. A pull request to update that doc would be welcome :)

**Reply by fabtjar** [reply]
Done! https://github.com/wagtail/wagtail/pull/7599
